### PR TITLE
Try to fix E2E tests

### PR DESCRIPTION
**Pull request by leomayleomay**
Closes

other changes

- [ ] The impure tests via `nix run .#offchain:test` were run and all are passing.
- [ ] Where applicable, grammar, punctuation, and spelling within code and comments should be plausibly correct.
- [ ] The issue(s) that the work is addressing should be linked in the PR description.
- [ ] Any significant changes beyond what's clear from the linked issue(s) are documented in the PR description

### CODE REVIEW

File: offchain/hello-world-browser/test/src/NamiWallet.purs [hunk 1]
```diff
@@ -104,43 +155,171 @@ runE2ETest
...
+    js = fold
+      [ "let input = $('" <> unwrap selector <> "').get(0);"
+      , "var nativeInputValueSetter = "
+          <>
+            " Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, 'value').set;"
+      , "nativeInputValueSetter.call(input, '" <> value <> "');"
+      , "var ev2 = new Event('input', { bubbles: true});"
+      , "input.dispatchEvent(ev2);"
+      ]
+  void $ T.unsafeEvaluateStringFunction js page
```
[the remark this answers is not in the document]
Comment (Geometer1729) [reply offchain/hello-world-browser/test/src/NamiWallet.purs:278]: Sounds good. 
I wonder why they didn't use FFI, I'll ask in the discord.

**Comment by epiro-ps** [comment]
@leomayleomay Given that this project is now open-source, it might be best to move forward without the faucet until we can securely version the API key. This will also unblock https://github.com/ArdanaLabs/dUSD/pull/349.

File: offchain/hello-world-api/src/Api.purs [hunk 2]
```diff
@@ -124,14 +124,18 @@ redeemFromScript
...
+  key <- liftContractM "no wallet" =<< ownPaymentPubKeyHash
   let
     lookups :: Lookups.ScriptLookups PlutusData
     lookups = Lookups.validator validator
       <> Lookups.unspentOutputs utxos
 
     constraints :: TxConstraints Unit Unit
     constraints = Constraints.mustSpendScriptOutput txInput spendRedeemer
-  _ <- buildBalanceSignAndSubmitTx lookups constraints
+      <> Constraints.mustBeSignedBy key
```
Comment (epiro-ps) [inline offchain/hello-world-api/src/Api.purs:135]: Could we add a todo here to remove this workaround once CTL/Nami resolves https://github.com/Plutonomicon/cardano-transaction-lib/issues/1079?

**Review by epiro-ps — CHANGES_REQUESTED** [review]

File: offchain/hello-world-api/src/Api.purs (hunk 2 above)
Comment (Geometer1729) [reply offchain/hello-world-api/src/Api.purs:139]: :+1: @leomayleomay I pushed this Todo cause I should've added it in the first place and it shouldn't conflict you.

Comment (leomayleomay) [reply offchain/hello-world-api/src/Api.purs:139]: :+1:

**Pull request merged**